simphony_mayavi/sources/slim_cuds_source.py

```diff
@@ -0,0 +1,623 @@
+    def _data_changed(self, old, new):
+        """When the vtk dataset changes, adjust the pipeline
+        accordingly so that we sync against the new dataset"""
+
+        if has_attributes(self.data):
+            aa = self._assign_attribute
+            self.configure_input_data(aa, new)
+            self._update_vtk_dataset_content()
+            aa.update()
+            self.outputs = [aa.output]
+        else:
+            self.outputs = [self.data]
+
+        # Notify the pipeline to refresh against the new data.
+        self.data_changed = True
+
+        self.output_info.datasets = \
+            [get_tvtk_dataset_name(self.outputs[0])]
+
+        # Add an observer to the VTK dataset after removing the one
+        # for the old dataset.  We use the messenger to avoid an
+        # uncollectable reference cycle.  See the
+        # tvtk.messenger module documentation for details.
+        if old is not None:
+            old.remove_observer(self._observer_id)
+        self._observer_id = new.add_observer('ModifiedEvent',
+                                             messenger.send)
+        new_vtk = tvtk.to_vtk(new)
+        messenger.connect(new_vtk,
+                          'ModifiedEvent',
+                          self._fire_data_changed)
+
+    # Private
+    # -------------------------------------------------------------------------
+
+    def _do_full_refresh(self):
+        """Performs appropriate refresh against the current cuds data,
+        and chooses the appropriate defaults considering the current
+        selected combobox names (if present), or the defaults specified
+        at construction"""
+
+        current_names = self._collect_current_names()
+
+        self._fill_datatype_enums()
+        self._select_names_silently(
+            default_names=self._constructor_default_names,
+            current_names=current_names,
+        )
+        self._update_vtk_cuds_from_cuds()
+
+    def _fill_datatype_enums(self):
+        """Fills the "comboboxes" enumeration options from the current cuds.
+        Works appropriately if the cuds is None.
+
+        Note that after filling the comboboxes, the selected entry is enforced
+        to the empty selection "". This is because we want to postpone this
+        selection to have more flexibility and reduce wasted setting
+        as much as possible.
+        """
+        cuds = self.cuds
+        available_keys = _available_keys(cuds) if cuds is not None else {}
+
+        # We go through the individual combobox lists, computing their names
+        # and using reflection, populating each _list with the keys available,
+        # adding a space for "no selection" and finally setting the appropriate
+        # default in the _name trait
+        for data_type, attr in data_type_attrs():
+            data_type_attr = "{}_{}".format(data_type, attr)
+            keys = available_keys.get(data_type_attr, set())
+            entries = sorted([key.name for key in keys])
```

Comment:
I agree.  Not sure if this slight difference in behaviour is worth a mention in the init docstring.